### PR TITLE
Support FindPython.

### LINKED ISSUE #506: FindPython support
I am currently working on adding full CMake 3.12+'s FindPython support to pybind11. Would it be possible to support for FindPython in Scikit-Build's extensions, too?

### PR DESCRIPTION
**Pull request by hameerabbasi**
Closes #506 and closes #516 and closes #553.

### CODE REVIEW

File: tests/samples/hello-numpy/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+include(GNUInstallDirs)
+
+find_package(Python REQUIRED COMPONENTS Development NumPy)
+
+# Fetch pybind11
+include(FetchContent)
+
+FetchContent_Declare(
+  pybind11
+  URL https://github.com/pybind/pybind11/archive/refs/tags/v2.6.2.tar.gz
```
Comment (henryiii) [inline tests/samples/hello-numpy/CMakeLists.txt:15]: Guessing this (and the ...3.18 above) were copied from another example? We really should be update the pins/limits. Doesn't have to be in this PR, though, just noting for later.

File: tests/samples/hello-numpy/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+         pybind11::module
+)
+
+target_compile_definitions(${python_module_name}
+                           PRIVATE VERSION_INFO=${EXAMPLE_VERSION_INFO})
+
+install(TARGETS ${python_module_name} DESTINATION .)
+
+# Quiet a warning, since this project is only valid with SKBUILD
+set(ignoreMe "${SKBUILD}")
```
Comment (henryiii) [inline tests/samples/hello-numpy/CMakeLists.txt:37]: Technically, we now have a _lot_ of variables that are going to produce warnings (this is why I haven't applied this sort of a fix yet. But I think it's fine, and I have a long term idea for fixing these warnings). Not sure it makes sense to just quiet this one anymore. But again, not for this PR.

File: tests/test_numpy.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+BASE = os.path.dirname(DIR)
+
+
+def test_pep518(pep518):
+    subprocess.check_call([sys.executable, "-m", "build", "--wheel"], cwd=HELLO_NUMPY)
+
+
+def test_dual_pep518(pep518):
+    subprocess.check_call([sys.executable, "-m", "build", "--wheel"], cwd=HELLO_NUMPY)
+    subprocess.check_call([sys.executable, "-m", "build", "--wheel"], cwd=HELLO_NUMPY)
```
Comment (henryiii) [inline tests/test_numpy.py:16]:
```suggestion
```

I don't think we need this test in this case, just making sure it works is fine (and these tests are expensive).

File: tests/test_numpy.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import os
+import subprocess
+import sys
+
+DIR = os.path.dirname(os.path.abspath(__file__))
+HELLO_NUMPY = os.path.join(DIR, "samples/hello-numpy")
+BASE = os.path.dirname(DIR)
+
+
+def test_pep518(pep518):
```
Comment (henryiii) [inline tests/test_numpy.py:10]:
```suggestion
def test_pep518_findpython(pep518):
```

**Comment by hameerabbasi** [comment]
I've gone through and applied the simple fixes suggested in the review.

**Review by henryiii — APPROVED** [review]
I think this is good. We sadly will produce unused variable warnings, but that's unavoidable for now, and I think I have a general solution to be applied later. I think we should get this in so we get it into a final Python 2 release. Thanks!

File: CHANGES.rst [hunk 5]
```diff
@@ -5,6 +5,20 @@ Release Notes
 This is the list of changes to scikit-build between each release. For full
 details, see the commit logs at http://github.com/scikit-build/scikit-build
 
+Scikit-build 0.15.0
+===================
+
+This release is the final (again) release for Python < 3.6 and MSVC<2017. Support
+for FindPython from CMake 3.12+ was added, inluding FindPython2.
```
Comment (henryiii) [inline CHANGES.rst:12]:
```suggestion
for FindPython from CMake 3.12+ was added, including FindPython2.
```

**Comment by henryiii** [comment]
We'll need to rebase #485 on this, so let's get it in. Thank you!

**Pull request merged**